### PR TITLE
Rewrite software review post

### PR DESCRIPTION
**Pull request by noamross**
Major rewrite based on @karthik's feedback and reviewer survey.

### CODE REVIEW

**Comment by karthik** [comment]
Thanks @noam. Will read over and post comments by tomorrow.

**Comment by cboettig** [comment]
@noamross This looks really great.  It may be beyond the scope of this post, but I kinda wonder if it would make sense to link this to some of the previous discussion/debate about code review in the academic context.  

For instance, I wonder if it would be good to bring up, at least in passing, the Mozilla Code Review exercise that was [covered in Nature](http://doi.org/10.1038/501472a) a few years ago.  At the time, Roger Peng & others gave some [strong push back](https://twitter.com/Erika_Check/status/382911015358181376/photo/1) that a wider adoption of code review would simply discourage researchers from publishing software.  

Contrary to that hypothesis, I think there's a strong case to be made that we have found our review process to _encourage_ publishing software.  This also ties into the broader question which I don't feel we try and speak to directly here: _why do people on-board their packages_?  Is it the same reason that people submit to CRAN, or different?  

Is there any lesson a journal editor at a place that publishes lots of software papers, like MEE, can take away from this blog post on our experience? (Perhaps @richfitz can weigh in based on his experience there).  Just a thought.

File: _drafts/2016-03-22-software-review.Rmd [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+  - software
+  - review
+  - infrastructure
+---
+
+Code review, in which peers manually inspect the source code of software
+written by others, is widely recognized as one of the best tools for finding
+bugs in software. Code review is relatively uncommon in *scientific* software
+development, though. Scientists, despite being familiar with the process of
+peer review, often have little exposure code review due to lack of training and
```
Comment (sckott) [inline _drafts/2016-03-22-software-review.Rmd]: _little exposure `to` code review_

File: _drafts/2016-03-22-software-review.Rmd [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+historically little incentive to share the source code from their research. So
+scientific code, from one-off scripts to reusable R packages, is rarely subject
+to review. Most R packages are subject only to the automated checks required by
+CRAN, which primarily ensure that packages can be installed on multiple systems.
+As such, The burden is on software users to discern well-written and efficient
+packages from poorly written ones.
+
+ROpenSci is a community of developer-scientists, creating R packages for other
```
Comment (sckott) [inline _drafts/2016-03-22-software-review.Rmd]: s/ROpenSci/rOpenSci/

File: _drafts/2016-03-22-software-review.Rmd [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+scientists, and our package contributors have a mix of backgrounds. We aim to
+serve our users with high-quality software, but also promote best practices
```
Comment (sckott) [inline _drafts/2016-03-22-software-review.Rmd]: the `but` here seems to put high quality software and promoting best practices at odds, or maybe i'm not reading it right

File: _drafts/2016-03-22-software-review.Rmd [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+ROpenSci's package review process owes much to the experiments of others
+(such as the [Mozilla Science Lab](https://mozillascience.org/code-review-
+for-science-what-we-learned)), as well as the [active feedback](https://
+discuss.ropensci.org/t/code-review-onboarding-milestones/180) [from our
+community](https://discuss.ropensci.org/t/how-could-the-onboarding-package-
+review-process-be-even-better/302). Here's how it works: When an author submits
+a package, our editors evaluate it for fit according to our [criteria](https://
+github.com/ropensci/policies#package-fit), then assign reviewers who evaluate
+the package for usability, quality, and style based on our [guidelines](https://
+github.com/ropensci/packaging_guide#ropensci- packaging-guide). After the
```
Comment (sckott) [inline _drafts/2016-03-22-software-review.Rmd]: link has a space in it , shoudl be https://github.com/ropensci/packaging_guide#ropensci-packaging-guide

File: _drafts/2016-03-22-software-review.Rmd [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+
+### We can be clearer about the beginning and end of the process
+
+> "It wasn't immediately clear what to do"
+    
+A few respondents pointed out we could be better at explaining the review
+process, both in how to get started and how it is supposed to wrap up. For the
+former, we've recently updated our [reviewer guide](https://github.com/ropensci/
+onboarding/wiki/For-Reviewers), including adding links to previous reviews. We
+hope as our reviewer pool gets more experienced, and as as software reviews become
```
Comment (sckott) [inline _drafts/2016-03-22-software-review.Rmd]: remove 2nd `as`

File: _drafts/2016-03-22-software-review.Rmd [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+more common, this gets easier. However, as our pool of editors and reviewers
+grows, we'll need to ensure that our communication is clear.
+    
+As for the *end* of the review, this can be an area of considerable ambiguity.
+There's a clear endpoint when a package is accepted, but with no "rejections"
+some reviewers weren't sure how to respond if authors didn't follow up on their
+comments. We realize it can be demotivating to reviewers if their suggestions
+aren't acted upon. (One reviewer pointed out that seeing her suggestions
+implemented as a positive motivator.) As such, it may be worthwhile to push
+authors to make a some commitment on submitting their package that they will to
```
Comment (sckott) [inline _drafts/2016-03-22-software-review.Rmd]: sentence _As such...motivating deadline_ doesn't make much sense

File: _drafts/2016-03-22-software-review.Rmd [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+
+[^RefC]: Also, "RefClasses are
+the devil", said one reviewer.  Interpret that as you may
+
+# Questions and ideas for the future.
+
+- *Scaling and reviewer incentives.* Like academic paper review or contributing
+to free open-source projects, our package review is a volunteer activity.
+How do we build an experienced reviewer base, and still maintain enthusiasm
+avoid overburdening anyone? We will have need to expand our reviewer pool in
```
Comment (sckott) [inline _drafts/2016-03-22-software-review.Rmd]: and avoid... ?

Comment (sckott) [inline _drafts/2016-03-22-software-review.Rmd]: we will need to expand

File: _drafts/2016-03-22-software-review.Rmd [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+contributorship-badges), or providing an easier route to publishing software papers.
+
+- *Automation.* How can we automate more parts of the review process so as to
+get more value out of reviewer and reviewees time? One suggestion has been to submit packages
+[*as* pull requests](https://discuss.ropensci.org/t/how- could-the-onboarding-
+package-review-process-be-even-better/302/3) to take more advantage of GitHub
+review features such as in-line commenting. This may allow us to move the burden
+of setting up continuous integration and testing away from the authors and onto
+our own pipeline, and allow us to add ROpenSci-specific tests. We've also started using [automated reminders](https://
+github.com/ropenscilabs/heythere) to keep up with reviewers, which reduces the
```
Comment (sckott) [inline _drafts/2016-03-22-software-review.Rmd]: link is broken over 2 lines

**Comment by noamross** [comment]
@cboettig I didn't want to dive too much into the weeds of that discussion, but I added this:

> Interestingly, no one mentioned the credential of an rOpenSci "badge" as a
positive aspect of review.  While the badge may be a motivating factor,
it seems from the responses that authors primarily value in the feedback itself.
There has been some [argument](http://simplystatistics.org/2013/09/26/how-could-code-review-discourage-code-disclosure-reviewers-with-motivation/) whether code
review will encourage or discourage scientists to publish their
code.  While our package authors represent a specific subset of scientists - those knowledgeable and motivated enough to create and disseminate packages - we think
our pilot shows that a well-designed review process can be encouraging.

**Comment by sckott** [comment]
@karthik you said you'd give some comments. still want to do that? 

@cboettig any other thoughts

**Comment by karthik** [comment]
Post is fantastic! Like amazeballs. 
/me tips hat to Noam.

@sckott Nothing more from me. Ready to post.

**Comment by noamross** [comment]
Thanks @karthik! Your feedback was really helpful.

**Pull request merged**